### ISSUE TITLE
vcpkg update check keeps warning to update after running bootstrap.

### ISSUE THREAD

**Issue by Farwaykorse** [Microsoft/vcpkg#4586]
Vcpkg keeps reporting an update to be available. After running `.\bootstrap-vcpkg.bat`.

``vcpkg update`` returns:
``Warning: Different source is available for vcpkg (0.0.113 -> 2018.10.20). Use .\bootstrap-vcpkg.bat to update.``

There were no build errors.

**Comment by ras0219-msft** [comment]
Thanks for the bug report!

Could you confirm which vcpkg you're using and what the local version is?
```
D:\src\vcpkg> vcpkg version
Vcpkg package management program version 2018.10.20

See LICENSE.txt for license information.
D:\src\vcpkg> where.exe vcpkg
D:\src\vcpkg\toolsrc\x64\Release\vcpkg.exe
D:\src\vcpkg> type .\toolsrc\VERSION.txt
"2018.10.20"
```

**Comment by Farwaykorse** [comment]
On Win10 with VS2017, and a WSL install using the same directory.
Only the windows version shows this issue.
````
C:\src\vcpkg [master ≡]> git pull
Already up to date.
C:\src\vcpkg [master ≡]> .\bootstrap-vcpkg.bat

Building vcpkg.exe ...
....
  vcpkglib.vcxproj -> C:\src\vcpkg\toolsrc\Build\Win32\Release\vcpkglib.lib
  vcpkg.cpp
  Generating code
  All 15263 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  vcpkg.vcxproj -> C:\src\vcpkg\toolsrc\Build\Win32\Release\vcpkg.exe
  vcpkgmetricsuploader.cpp
  Generating code
  All 5401 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  vcpkgmetricsuploader.vcxproj -> C:\src\vcpkg\toolsrc\Build\Win32\Release\vcpkgmetricsuploader.exe

Building vcpkg.exe... done.
C:\src\vcpkg [master ≡]> vcpkg update
Warning: Different source is available for vcpkg (0.0.113 -> 2018.10.20). Use .\bootstrap-vcpkg.bat to update.
Using local portfile versions. To update the local portfiles, use `git pull`.
No packages need updating.
C:\src\vcpkg [master ≡]> vcpkg version
Vcpkg package management program version 0.0.113-nohash

See LICENSE.txt for license information.
C:\src\vcpkg [master ≡]> where.exe vcpkg
C:\src\vcpkg\vcpkg
C:\src\vcpkg\vcpkg.exe
C:\src\vcpkg [master ≡]> type .\toolsrc\VERSION.txt
"2018.10.20"
````

After looking at this, I noticed that while the executable for Linux (`C:\src\vcpkg\vcpkg`) has a recent modification date, the Windows executable (`C:\src\vcpkg\vcpkg.exe`) was last modified at May 22.
New builds are now generated in `.\toolsrc\Build\Win32\Release`.
While the old build-files (dated in May) remain in `.\toolsrc\Release`.

**Comment by alexkaratarakis** [comment]
The problem here is that the build is happening in `.\toolsrc\Build\Win32\Release`.
We expect it to be at .\toolsrc\Release.
We do a build, and then we copy the file to the vcpkg root after the build:

```
Copy-Item "$vcpkgReleaseDir\vcpkg.exe" "$vcpkgRootDir\vcpkg.exe"
```

It looks like there is an old build in `\toolsrc\Release`, so in your case, the script is doing a build in another dir, completely ignores that dir and then copies vcpkg.exe from the old build. 

I am wondering why your build directory is what it is, but regardless: please try this branch
https://github.com/Microsoft/vcpkg/tree/dev/alkarata/bootstrapOutDir
It explicitly sets the output directory which should render us unaffected to issues like this.

**Comment by alexkaratarakis** [comment]
Related: #3641 
@aalpert-bluescape This also addresses your case-sensitivity issue. Please let me know if it indeed does.

**Comment by aalpert-bluescape** [comment]
@alexkaratarakis Doesn't look like this branch helps for that issue. I'll reproduce the error message I get:

```
Building vcpkg.exe ...

  pch.cpp
CL : fatal error C1083: Cannot open compiler intermediate file: 'c:\dev\vcpkg\toolsrc\vcpkglib\release\vcpkglib.pch': No such file or directory [C:\dev\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj]
```

The problem is that it's looking for 'c:\dev\vcpkg\toolsrc\vcpkglib\release\vcpkglib.pch', when the file is 'C:\dev\vcpkg\toolsrc\vcpkglib\Release\vcpkglib.pch'. Your branch appears to only change the output dir for the vcpkg.exe built executable to 'toolsrc\msbuild.rel.x86', but I'm still seeing the intermediate files in 'toolsrc\vcpkglib\Release'.

**Comment by alexkaratarakis** [comment]
@aalpert-bluescape 
I missed the intermediate paths. Alright, I modified the vcxproj files to explicitly specify output AND intermediate path. Since I am doing that, I made the build be out-of-source. I have pushed the changes to the same same branch.

https://github.com/Microsoft/vcpkg/tree/dev/alkarata/bootstrapOutDir

@Farwaykorse Please let me know if the branch fixes your issue.

I made the output path in vcxproj the same as the OutDir override in bootstrap so if it does NOT fix the issue, I will remove the OutDir parameter and look into ignoring props files, which I suspect are the cause of this.

**Comment by Farwaykorse** [comment]
@alexkaratarakis the branch `dev/alkarata/bootstrapOutDir` @ commit f89621e works for me.
Though I did notice double backslashes in the paths.
````
  vcpkglib.vcxproj -> C:\src\vcpkg\toolsrc\msbuild.x86.release\\vcpkglib.lib
[...]
  vcpkg.vcxproj -> C:\src\vcpkg\toolsrc\msbuild.x86.release\\vcpkg.exe
[...]
  vcpkgmetricsuploader.vcxproj -> C:\src\vcpkg\toolsrc\msbuild.x86.release\\vcpkgmetricsuploader.exe
````
I found the reason for my issue (the paths did seem somewhat familiar...).
I've modified my default output and intermediate directories in `Microsoft.Cpp.(x64/Win32).user.props` to use the `Build\$(Platform)` directory.

Hardening dependencies should work against this.
Or maybe better, unless you want to offer this kind of customisation, would be to remove the inclusion of user defined default properties from the .vcxproj files.
`````
<Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
`````

**Comment by aalpert-bluescape** [comment]
@alexkaratarakis With f89621e59f8287b7deab5fc961252d2b133b0f33 it is building for me now, so that looks to have fixed the case-sensitivity issue.

**Comment by alexkaratarakis** [comment]
All fixed after 5a1a75d2d6bb27675bca2f009c7ae5579e84cea7.
Thanks @ all :).
